packages/dev/core/src/FlowGraph/flowGraphPath.ts

```diff
@@ -0,0 +1,82 @@
+import type { FlowGraphContext } from "./flowGraphContext";
+
+const SEPARATORS = /\/|\./;
+/*
+ * This class represents a path of type /x/{y}/z/.../w that is evaluated
+ * on a target object. The string between curly braces ({y} in the example)
+ * is a special template string that is replaced during runtime.
+ */
+export class FlowGraphPath {
+    private path: string;
+    private templateSubstitutions: {
+        [key: string]: number;
+    } = {}; // this is a map of template strings to values that are substituted during runtime
+    public hasTemplateStrings: boolean = false;
+
+    constructor(path: string) {
+        this.path = path;
+        const templateStrings = this._getTemplateStringsInPath(path);
+        this.hasTemplateStrings = templateStrings.length > 0;
+        for (const templateString of templateStrings) {
+            // Part of the path enclosed in curly braces = template string, we'll set a default
+            // value on the template substitutions for now
+            this.templateSubstitutions[templateString] = -1;
+        }
+    }
+
+    getTemplateStrings(): string[] {
+        return Object.keys(this.templateSubstitutions);
+    }
+
+    private _getTemplateStringsInPath(path: string): string[] {
+        const splitPath = path.split(SEPARATORS).filter((part) => part !== "");

```

Comment:
Maybe you could use a regexp that would parse the whole syntax, and not only the separators?

Reply:
If speed is important, then you should not even do a split and simply scan the string character by character (maybe it's the best solution if the code is executed a lot of time).

Reply:
This regexp: `/{(.*?)}/` is ungreedy matcher for all strings inside `{}`. Depending how you want to run the substitution, you could use it in different ways. using `String.prototype.matchAll` will probably give you everything you need to run the substitutions. It also supports empty template placeholders (i.e. {})